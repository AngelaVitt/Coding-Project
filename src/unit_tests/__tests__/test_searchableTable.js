import ReactTable from 'react-table'
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';

const columns = [{
  Header: 'IP Address',
  accessor: 'ipAddress' // String-based value accessors!
}]

const data = [{
    ipAddress: 34.23423
  },{
    ipAddress: 54.23423
  }]


export default class SearchableTable extends React.Component {
  

  render() {
    return (
      <ReactTable 
          data = {data}
          columns = {columns}
          sortable = {false}
          filterable
          defaultPageSize = {10}
          pageSizeOptions = {[5, 10, 20, 25, 50, 100, 500, 1000]}
          defaultFilterMethod = {(filter, row, column) => {
            const id = filter.pivotId || filter.id
            return row[id] !== undefined ? String(row[id]).toLowerCase().startsWith(filter.value.toLowerCase()) : true
          }}
          >
        
      </ReactTable>
    );
  }
}


test("renders correctly", () => {
  const renderer = new ShallowRenderer();

  const result = renderer.render(<SearchableTable/>);

  expect(result).toMatchSnapshot();
});
