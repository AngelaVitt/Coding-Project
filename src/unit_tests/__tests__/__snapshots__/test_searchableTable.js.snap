// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ReactTable
  AggregatedComponent={[Function]}
  ExpanderComponent={[Function]}
  FilterComponent={[Function]}
  LoadingComponent={[Function]}
  NextComponent={undefined}
  NoDataComponent={[Function]}
  PadRowComponent={[Function]}
  PaginationComponent={[Function]}
  PivotComponent={undefined}
  PivotValueComponent={[Function]}
  PreviousComponent={undefined}
  ResizerComponent={[Function]}
  TableComponent={[Function]}
  TbodyComponent={[Function]}
  TdComponent={[Function]}
  TfootComponent={[Function]}
  ThComponent={[Function]}
  TheadComponent={[Function]}
  TrComponent={[Function]}
  TrGroupComponent={[Function]}
  aggregatedKey="_aggregated"
  className=""
  collapseOnDataChange={true}
  collapseOnPageChange={true}
  collapseOnSortingChange={true}
  column={
    Object {
      "Aggregated": undefined,
      "Cell": undefined,
      "Expander": undefined,
      "Filter": undefined,
      "Footer": undefined,
      "Header": undefined,
      "Pivot": undefined,
      "PivotValue": undefined,
      "aggregate": undefined,
      "className": "",
      "filterAll": false,
      "filterMethod": undefined,
      "filterable": undefined,
      "footerClassName": "",
      "footerStyle": Object {},
      "getFooterProps": [Function],
      "getHeaderProps": [Function],
      "getProps": [Function],
      "headerClassName": "",
      "headerStyle": Object {},
      "minWidth": 100,
      "resizable": undefined,
      "show": true,
      "sortMethod": undefined,
      "sortable": undefined,
      "style": Object {},
    }
  }
  columns={
    Array [
      Object {
        "Header": "IP Address",
        "accessor": "ipAddress",
      },
    ]
  }
  data={
    Array [
      Object {
        "ipAddress": 34.23423,
      },
      Object {
        "ipAddress": 54.23423,
      },
    ]
  }
  defaultExpanded={Object {}}
  defaultFilterMethod={[Function]}
  defaultFiltered={Array []}
  defaultPageSize={10}
  defaultResized={Array []}
  defaultSortDesc={false}
  defaultSortMethod={[Function]}
  defaultSorted={Array []}
  expanderDefaults={
    Object {
      "filterable": false,
      "resizable": false,
      "sortable": false,
      "width": 35,
    }
  }
  filterable={true}
  freezeWhenExpanded={false}
  getLoadingProps={[Function]}
  getNoDataProps={[Function]}
  getPaginationProps={[Function]}
  getProps={[Function]}
  getResizerProps={[Function]}
  getTableProps={[Function]}
  getTbodyProps={[Function]}
  getTdProps={[Function]}
  getTfootProps={[Function]}
  getTfootTdProps={[Function]}
  getTfootTrProps={[Function]}
  getTheadFilterProps={[Function]}
  getTheadFilterThProps={[Function]}
  getTheadFilterTrProps={[Function]}
  getTheadGroupProps={[Function]}
  getTheadGroupThProps={[Function]}
  getTheadGroupTrProps={[Function]}
  getTheadProps={[Function]}
  getTheadThProps={[Function]}
  getTheadTrProps={[Function]}
  getTrGroupProps={[Function]}
  getTrProps={[Function]}
  groupedByPivotKey="_groupedByPivot"
  indexKey="_index"
  loading={false}
  loadingText="Loading..."
  multiSort={true}
  nestingLevelKey="_nestingLevel"
  nextText="Next"
  noDataText="No rows found"
  ofText="of"
  onExpandedChange={undefined}
  onFetchData={[Function]}
  onFilteredChange={undefined}
  onPageChange={undefined}
  onPageSizeChange={undefined}
  onResizedChange={undefined}
  onSortedChange={undefined}
  originalKey="_original"
  pageSizeOptions={
    Array [
      5,
      10,
      20,
      25,
      50,
      100,
      500,
      1000,
    ]
  }
  pageText="Page"
  pivotBy={undefined}
  pivotDefaults={Object {}}
  pivotIDKey="_pivotID"
  pivotValKey="_pivotVal"
  previousText="Previous"
  resizable={true}
  rowsText="rows"
  showPageJump={true}
  showPageSizeOptions={true}
  showPagination={true}
  showPaginationBottom={true}
  showPaginationTop={false}
  sortable={false}
  style={Object {}}
  subRowsKey="_subRows"
/>
`;
